laalaa/apps/advisers/geocoder.py

```diff
@@ -37,7 +38,16 @@ def result_to_postcode(result):
 
 def lookup_postcode(postcode):
     normalised_postcode = normalise_postcode(postcode)
-    raw = requests.get(
+
+    session = requests.Session()
+
+    retry_strategy = Retry(total=5, backoff_factor=0.1)
+
+    # `max_retries` only applies to failed DNS lookups, socket connections and connection timeouts, never to requests where data has made it to the server.
+    # by default, Requests does not retry failed connections. Please see documentation: https://docs.python-requests.org/en/latest/api/#requests.adapters.HTTPAdapter
```

Comment:
I'm not sure that this comment is completely correct given the use of Retry. 

Also completely confused about which Retry we should be using - here you're using request.adapters.Retry but I have been looking at the docs for 
https://urllib3.readthedocs.io/en/stable/reference/urllib3.util.html#urllib3.util.Retry ? 

Think we need a FALA branch to test this with?

Reply:
So I was basing this on [the stack overflow](https://stackoverflow.com/questions/15431044/can-i-set-max-retries-for-requests-request) provided on the Jira ticket. 

I agree that the provided documentation wasn't clear on which Retry we were using. So I looked to see what others did and found [this](https://github.com/ministryofjustice/operations-engineering/blob/f83e228b056eeb84e2722e429d89498a98df56bd/services/operations_engineering_reports.py#L4). Here they use the `urllib3` Retry. 

I agree we should test with FALA staging.